core/src/main/java/org/apache/iceberg/MetadataUpdate.java

```diff
@@ -165,6 +165,24 @@ public String name() {
     }
   }
 
+  class CreateSnapshotRef implements MetadataUpdate {
```

Comment:
A rename/update to a ref will be an ordered Remove + Create. Seems a bit inefficient in the sense of how many updates are needed to encapsulate a single update, but seems like the intention of this metadata update is to give an ordered, immutable series of operations to build metadata state.

Reply:
I see the rationale for this, but I don't think there is a reason to have separate `CreateSnapshotRef` and `SetSnapshotRef` classes. I think we should just add the ref metadata to the `SetSnapshotRef` class. `SetSnapshotRef` was intended to be an idempotent update that creates or updates a ref.

Reply:
Ah I missed that we already had a SetSnapshotRef. Considering that it's idempotent, then yes, we should just add the ref information to it.